### PR TITLE
 nixos/no-x-libs: fix some gst plugins

### PR DESCRIPTION
**Pull request by SuperSandro2000**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/libraries/gstreamer/rs/default.nix [hunk 1]
```diff
@@ -120,6 +121,11 @@ let
       "livesync"
       "fallbackswitch"
       "togglerecord"
+    ] ++ lib.optionals (!withGtkPlugins) [
+      "fallbackswitch"
+      "gtk4"
+      "livesync"
+      "togglerecord"
```
Comment (MinerSebas) [inline pkgs/development/libraries/gstreamer/rs/default.nix:128]: Couldn't this be merged with the conditon above to `!gst-plugins-base.glEnabled || !withGtkPlugins` as they contain the same four entries.

**Review by Izorkin — APPROVED** [review]

File: pkgs/development/libraries/gstreamer/rs/default.nix [hunk 2]
```diff
@@ -114,7 +115,7 @@ let
       "reqwest" # tests hang on darwin
       "threadshare" # tests cannot bind to localhost on darwin
       "webp" # not supported on darwin (upstream crate issue)
-    ] ++ lib.optionals (!gst-plugins-base.glEnabled) [
+    ] ++ lib.optionals (!gst-plugins-base.glEnabled || withGtkPlugins) [
```
Comment (MinerSebas) [inline pkgs/development/libraries/gstreamer/rs/default.nix:118]: I think this is a copy-paste error.
```suggestion
    ] ++ lib.optionals (!gst-plugins-base.glEnabled || !withGtkPlugins) [
```

File: nixos/modules/config/no-x-libs.nix [hunk 3]
```diff
@@ -46,8 +46,9 @@ with lib;
       graphviz = super.graphviz-nox;
       gst_all_1 = super.gst_all_1 // {
         gst-plugins-bad = super.gst_all_1.gst-plugins-bad.override { guiSupport = false; };
-        gst-plugins-base = super.gst_all_1.gst-plugins-base.override { enableWayland = false; enableX11 = false; };
+        gst-plugins-base = super.gst_all_1.gst-plugins-base.override { enableGl = false; enableWayland = false; enableX11 = false; };
         gst-plugins-good = super.gst_all_1.gst-plugins-good.override { enableWayland = false; enableX11 = false; gtkSupport = false; qt5Support = false; qt6Support = false; };
+        gst-plugins-rs = super.gst_all_1.gst-plugins-rs.override { withGtkPlugins = false; };
```
Comment (MinerSebas) [inline nixos/modules/config/no-x-libs.nix:51]: Nitpick, but `gst-plugins-good` has a "feature" called `gtkSupport`, but the new feature for `gst-plugin-rs` calls it `withGtkPlugins`.

Comment (SuperSandro2000) [reply nixos/modules/config/no-x-libs.nix:51]: That's because it disabled 4 plugins. gtkSupport would more likely translate to just the gtk plugin.

**Review by MinerSebas — APPROVED** [review]

**Pull request merged**